README.md

```diff
@@ -8,60 +9,77 @@
-
-# How to Install
-
-## Centos 7
-
-1. Import the public GPG key (replace `0.0.1-65` with the release you want)
-
-```
-rpm --import https://github.com/xmidt-org/caduceus/releases/download/0.0.1-65/RPM-GPG-KEY-comcast-xmidt
-```
-
-2. Install the rpm with yum (so it installs any/all dependencies for you)
-
-```
-yum install https://github.com/xmidt-org/caduceus/releases/download/0.0.1-65/caduceus-0.0.1-65.el6.x86_64.rpm
+## Summary
+The [XMiDT](https://xmidt.io/) server for delivering events from talaria to the
+registered consumer. Refer the [overview docs](https://xmidt.io/docs/introduction/overview/)
+for more information on how caduceus fits into the overall picture.
+
+## Details
+Caduceus has one function: to deliver events to a consumer.
+To enable this caduceus has three endpoints: 1) receive events, 2) register webhooks,
+and 3) get webhooks.
+
+#### Notify - `api/v3/notify` endpoint
+The notify endpoint will accept a `msgpack` encoding of a [WRP Messages](https://github.com/xmidt-org/wrp-c/wiki/Web-Routing-Protocol).
+If a webhook is registered and matches the device regex and event regex the event
+will be sent to the corresponding url. To register a webhook refer to
+the [webhook registration section](#notify)
+
+#### Webhook - `/hook` endpoint
+To register a webhook, the consumer must send an http POST request to caduceus
+including the http url for receiving the events and a list of regex filters to get events.
+The following is an example request. Note: this is not a valid json since I added comments.
 ```
+{
+  "config" : {
+    # The url to push events to.
+    "url" : "http://localhost:8080/webhook",
+
+    # The content type event.
+    # (Optional) defaults to msgpack.
+    "content_type" : "application/json",
+
+    # The secret used for SHA1 HMAC.
+    # Used to validate the received payload.
+    # (Optional) defaults to no sha1 hmac.
+    "secret" : "secret",
+
+    # On failure how many times to retry the url for the event.
+    # (Optional) defaults to the configured value on the server
+    "max_retry_count": 3,
+
+    # Alternative urls to send requests too. A list of server urls to use in
+    # round robin for sending events.
+    # (Optional) defaults to no alternative urls.
+    "alt_urls" : [
+      "http://localhost:8080/webhook"
+    ]
+  },
 
-## Dockerized Caduceus
-Docker containers make life super easy.
-
-You will need SNS(or mock SNS) in order for Caduceus to work correctly.
-
-### Installation
-- [Docker](https://www.docker.com/) (duh)
-  - `brew install docker`
-
-</br>
+  # The URL to notify when we cut off a client due to overflow.
+  # (Optional) defaults no notification of error.
+  "failure_url" : "http://localhost:8080/webhook-failure",
 
-### Running
-#### Build the docker image
-```bash
-docker build -t caduceus:local .
-```
-This `build.sh` script will build the binary and docker image
+  # The list of regular expressions to match event type against.
+  "events" : [
+    ".*"
```

Comment:
I don't think we want to encourage anyone to register for all events? Especially if they take this and use it for anything in production...should we add some kind of warning?